### ISSUE TITLE
CSS property setting syntax does not permit the use of a colon (:) in the assigned value.

### ISSUE THREAD

**Issue by dbolack-ab** [naturalcrit/homebrewery#3886]
### Renderer

v3

### Browser

Chrome

### Operating System

Windows

### What happened?

In writing a how-to I discovered I was unable to assign a string value to a CSS property if it contained a `:` via the `{{}}` style setting syntax.

### Code

```gfm
{{footer,--chapterName:"'1:How-to'"}}
```

Generates 
```
<span class="inline-block footer" style="--chapterName:;">“‘1:How-to’”</span>
```

Instead of 

```
<span class="inline-block footer" style="--chapterName:‘1:How-to’;”></span>
```

**Comment by dbolack-ab** [comment]
The semi-colon is also superfluous.

**Comment by 5e-Cleric** [comment]
Right, we need to encode it for the content property, it has its own syntax.

**Comment by 5e-Cleric** [comment]
Right, we need to encode it for the content property, it has its own syntax.This also applies to html attributes.

**Comment by dbolack-ab** [comment]
> Right, we need to encode as a CSS string: https://developer.mozilla.org/en-US/docs/Web/CSS/string

I am unable to discern how my examples fail here?

**Comment by ericscheid** [comment]
(The semicolon being inserted here is expected as a separator of multiple style rules.)

The bug here is in _our_ parsing rules .. the colon is confusing our regex. Similar things happen with an equals `=`.

**Comment by dbolack-ab** [comment]
superceded by 4201
